src/Microsoft.OData.Core/ODataPropertyInfo.cs

```diff
@@ -0,0 +1,49 @@
+//---------------------------------------------------------------------
+// <copyright file="ODataPropertyInfo.cs" company="Microsoft">
+//      Copyright (C) Microsoft Corporation. All rights reserved. See License.txt in the project root for license information.
+// </copyright>
+//---------------------------------------------------------------------
+
+namespace Microsoft.OData
+{
+    using System.Collections.Generic;
+    using System.Diagnostics.CodeAnalysis;
+    using Microsoft.OData.Edm;
+
+    /// <summary>
+    /// Represents information about a single property of a resource.
+    /// </summary>
+    public class ODataPropertyInfo : ODataItem
+    {
+        /// <summary>Gets or sets the property name.</summary>
+        /// <returns>The property name.</returns>
+        public string Name
+        {
+            get;
+            set;
+        }
+
+        /// <summary>Gets or sets the kind of primitive type of the property.</summary>
+        /// <returns>The <see cref="EdmPrimitiveTypeKind"/> of the property.</returns>
+        public virtual EdmPrimitiveTypeKind PrimitiveTypeKind

```

Comment:
>EdmPrimitiveTypeKind [](start = 23, length = 20)

Why "EdmPrimitiveTypeKind"?

Reply:
Because PropertyInfo can be used to start writing any primitive property.

---
In reply to: [266659939](https://github.com/OData/odata.net/pull/1204#discussion_r266659939) [](ancestors = 266659939)

Reply:
It looks wired when i create an ODataProperty and Such property maybe not "primitive property".

---
In reply to: [266948740](https://github.com/OData/odata.net/pull/1204#discussion_r266948740) [](ancestors = 266948740,266659939)